test-fixtures/update-rust-component-version.py

```diff
@@ -1,129 +1,67 @@
     try:
-        repo = g.get_repo(GITHUB_REPO)
-        nightly_tag = find_latest_nightly_tag(repo.get_tags())
-        newest_tag = nightly_tag.name
-        newest_commit = str(nightly_tag.commit)
-        only_commit = re.findall(r'"([^"]*)"', newest_commit)
+        repo = Github().get_repo(GITHUB_REPO)
+        nightly_tag = repo.get_tags()[0]
+        return str(nightly_tag.name), re.findall(r'"([^"]*)"', str(nightly_tag.commit))[0]
     except requests.exceptions.HTTPError as err:
         raise SystemExit(err)
 
-    return (str(newest_tag), str(only_commit[0]))
 
-def find_latest_nightly_tag(tags):
-    for tag in tags:
-        if re.match(r'\d+\.0\.\d+', tag.name) is not None:
-            return tag
-    raise SyntaxError("No nightly tags found in rust-components-swift")
-
-def read_rust_components_tag_version():
-    # Read Package file to find the current rust-component version
+# Read the current version and commit of the Rust components from the SPM package file
+def read_rust_components_tag_version(SPM_PACKAGE):
     try:
-        f = open(SPM_PACKAGE)
-        data = json.load(f)
-
-        for i in data["pins"]:
-            if i["identity"] == "rust-components-swift":
-                # return the json with the RustComponent info
-                json_new_version = i["state"]
-        f.close()
-    except:
-        print("Could not read rust component tag")
-    # Return the current version and commit
-    finally:
-        return json_new_version["version"], json_new_version["revision"]
-
-def read_project_min_version():
-    line_number = 0
-    list_of_results = []
-    string_to_search = 'https://github.com/mozilla/rust-components-swift'
-
-    with open(FIREFOX_PROJECT) as f:
-        #if 'https://github.com/mozilla/rust-components-swift' in f.read():
-        line_read = 0
-        for line in f:
-            # For each line, check if line contains the string
-            line_number += 1
-            if string_to_search in line:
-                # If yes, then look for the field we are interested in: minimumVersion
-                for i in range(3):
-                    minimumVersion = next(f, '').strip()
-
-                version_found = minimumVersion.find("=")
-                last_line_position = minimumVersion.find(";")
-                current_tag_version = ''
-                # version format: XX.Y.Z
-                for i in range(version_found+2 , last_line_position):
-                    current_tag_version+= minimumVersion[i]
-                # Return the current rust-component version in project
-                return current_tag_version
-
+        with open(SPM_PACKAGE) as f:
+            data = json.load(f)
+            for i in data["pins"]:
+                if i["identity"] == "rust-components-swift":
+                    return i["state"]["version"], i["state"]["revision"]
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        print(f"Error reading rust component tag: {e}")
```

Comment:
We would need to change these print statment with `log.info` so that danger does not complain